### PR TITLE
[full-ci] [tests-only] Added api test for sharing spaces with/out manager permission

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
Added API tests for sharing spaces with and without `manager` permission.

Added scenarios:

1. `Scenario Outline: A user with editor or viewer role cannot share a space to another user`
2. `Scenario Outline: A user with manager role can share a space to another user`

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/4937 (share space without manager permission)

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSpacesShares/shareSpaces.feature [hunk 1]
```diff
@@ -87,3 +87,20 @@ Feature: Share spaces
     Then the user "Bob" should have a space called "share space" granted to "Bob" with role "viewer"
     And for user "Bob" the space "share space" should contain these entries:
       | test.txt |
+
+
+  Scenario Outline: An user without manager role cannot share a space to another user
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiSpacesShares/shareSpaces.feature:92]:
```suggestion
  Scenario Outline: user without manager role cannot share a space to another user
```

**Review by SwikritiT — COMMENTED** [review]
I see the pattern of naming the scenario in this file is something like `A user can share a space to another user` if only the manager can do these things maybe we can reword it to something like `user with manager role ....` as you've done with the newly added scenario?

**Review by SagarGi — COMMENTED** [review]
scenario LGTM

**Comment by grgprarup** [comment]
> I see the pattern of naming the scenario in this file is something like `A user can share a space to another user` if only the manager can do these things maybe we can reword it to something like `user with manager role ....` as you've done with the newly added scenario?

I have reworded the scenario `A user can share a space to another user` with `A Space Admin can share a space to another user`. Looking at the steps in this scenario, I think the test is targeted for Space Admin. @SwikritiT What do you think?

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Comment by SwikritiT** [comment]
> > I see the pattern of naming the scenario in this file is something like `A user can share a space to another user` if only the manager can do these things maybe we can reword it to something like `user with manager role ....` as you've done with the newly added scenario?
> 
> I have reworded the scenario `A user can share a space to another user` with `A Space Admin can share a space to another user`. Looking at the steps in this scenario, I think the test is targeted for Space Admin. @SwikritiT What do you think?

Looks good. Make the same changes in other scenario titles if it's needed otherwise, the added scenario looks good to me

**Comment by grgprarup** [comment]
@kiranparajuli589 @SwikritiT I have added another scenario for the `with manager role`. Please re-review them.

**Review by SwikritiT — COMMENTED** [review]
LGTM

File: tests/acceptance/features/apiSpacesShares/shareSpaces.feature [hunk 2]
```diff
@@ -101,3 +101,38 @@ Feature: Share spaces
...
+      | id        | %space_id%  |
+      | name      | share space |
+    Examples:
+      | role    |
+      | manager |
+      | editor  |
+      | viewer  |
+
+
+  Scenario Outline: A user without manager role cannot share a space to another user
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiSpacesShares/shareSpaces.feature:124]: aren't u checking use with various roles here? why saying only manger role?

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by kiranparajuli589 — APPROVED** [review]

**Review by ScharfViktor — APPROVED** [review]
LGTM
